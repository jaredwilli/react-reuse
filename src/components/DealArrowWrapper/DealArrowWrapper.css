
.deal {
    display: inline-flex;
    align-items: center;
}

.deal .deal-meta {
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin-left: 5px;
}
.deal .deal-meta h6 {
    color: #324054;
    font-size: 16px;
    text-transform: uppercase;
    white-space: nowrap;
    margin: 0;
}
.deal .deal-meta .imv {

}
.deal .deal-info {
    margin-left: 10px;
}
.deal .deal-info p {
    color: #324054;
    margin: 0;
}

.deal .deal-rating {
    background: none;
    border: none;
    padding-left: 0;
    padding-right: 0;
}

.deal .deal-rating svg.deal-arrow {
    vertical-align: middle;
    width: 31px;
}

.deal .deal-rating svg.deal-arrow.arrow-small {
    width: 21px;
}

.deal .deal-rating .uncertain {
    opacity: 0;
}
.deal .deal-rating .no-price {
    opacity: 0;
}
.deal .deal-rating .arrow {
    opacity: 1;
}

/* Great */
.deal .deal-rating.great svg.deal-arrow {
    transform: rotate(-90deg);
}
.deal .deal-rating.great circle {
    fill: #006b00;
}

/* Good */
.deal .deal-rating.good svg.deal-arrow {
    transform: rotate(-45deg);
}
.deal .deal-rating.good circle {
    fill: #009b00;
}

/* Fair */
.deal .deal-rating.fair svg.deal-arrow circle {
    fill: #00bd00;
}

/* High */
.deal .deal-rating.high svg.deal-arrow {
    transform: rotate(45deg);
}
.deal .deal-rating.high circle {
    fill: #ff8400;
}

/* Overpriced */
.deal .deal-rating.overpriced svg.deal-arrow {
    transform: rotate(90deg);
}
.deal .deal-rating.overpriced circle {
    fill: #f60000;
}

/* Uncertain */
.deal .deal-rating.uncertain svg.deal-arrow circle {
    fill: #ccc;
}
.deal .deal-rating.uncertain svg.deal-arrow .uncertain {
    opacity: 1;
}
.deal .deal-rating.uncertain svg.deal-arrow .arrow {
    opacity: 0;
}

/* No Price */
.deal .deal-rating.no-price svg.deal-arrow circle {
    fill: #ccc;
}
.deal .deal-rating.no-price svg.deal-arrow .no-price {
    opacity: 1;
}
.deal .deal-rating.no-price svg.deal-arrow .arrow {
    opacity: 0;
}
