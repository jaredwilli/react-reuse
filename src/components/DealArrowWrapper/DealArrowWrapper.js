import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import DealArrow from '../DealArrow/DealArrow';
import './DealArrowWrapper.css';

/** A wrapper around the deal arrow icons for styling and label/info output */
const DealArrowWrapper = (props) => {
    return (
        <div className="deal">
            {props.rating &&
                <div className={classnames('deal-rating', `${props.rating}`, `${props.className}`)}>
                    <DealArrow className={props.arrowClassName} />
                </div>
            }
            {(props.label || props.imv) &&
                <div className="deal-meta">
                    {props.label &&
                        <h6>{props.label}</h6>
                    }
                    {props.imv &&
                        <div className="deal-imv">
                            <p>{props.imv}</p>
                        </div>
                    }
                </div>
            }
            {props.info &&
                <div className="deal-info">
                    <p>{props.info}</p>
                </div>
            }
        </div>
    );
};

DealArrowWrapper.defaultProps = {
    className: '',
    label: '',
    imv: '',
    info: ''
};

DealArrowWrapper.propTypes = {
    /** The rating - great, good, fair, high, overpriced, outlier, overpriced */
    rating: PropTypes.string.isRequired,
    /** Any additional class name to add to the div */
    className: PropTypes.string,
    /** For small arrows provide the `arrow-small` class */
    arrowClassName: PropTypes.string,
    /** The label for the deal arrow - Great, Good, Fair etc. */
    label: PropTypes.string,
    /** The IMV text for the deal arrow */
    imv: PropTypes.string,
    /** The deal rating info text to display */
    info: PropTypes.string
};

export default DealArrowWrapper;
