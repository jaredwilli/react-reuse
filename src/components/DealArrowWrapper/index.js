import DealArrowWrapper from './DealArrowWrapper';

export default DealArrowWrapper;
