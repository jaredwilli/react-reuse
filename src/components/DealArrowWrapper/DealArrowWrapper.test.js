import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import DealArrowWrapper from './DealArrowWrapper';

describe('DealArrowWrapper', () => {
    it('renders the component correctly', () => {
        const tree = renderer.create(<DealArrowWrapper rating="great" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

});
