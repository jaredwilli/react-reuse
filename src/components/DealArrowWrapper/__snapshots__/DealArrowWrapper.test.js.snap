// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DealArrowWrapper renders the component correctly 1`] = `
<div
  className="deal"
>
  <div
    className="deal-rating great"
  >
    <svg
      className="deal-arrow"
      viewBox="0 0 34 34"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="17"
        cy="17"
        fill="#ccc"
        r="15"
      />
      <path
        className="arrow"
        d="M25.779,17l-7.97-7.97v4.857H8.221v6.226h9.588V24.97Z"
        fill="#fff"
      />
      <path
        className="no-price"
        d="M11.491,24.707a.731.731,0,0,1-.73-.733.723.723,0,0,1,.157-.451L21.934,9.571a.731.731,0,0,1,1.148.906h0L12.065,24.43A.725.725,0,0,1,11.491,24.707Z"
        fill="#fff"
      />
      <path
        className="uncertain"
        d="M12.713,10.43a7.843,7.843,0,0,1,.829-.635,5.98,5.98,0,0,1,.972-.522,6.1,6.1,0,0,1,1.134-.35A6.474,6.474,0,0,1,16.972,8.8a5.488,5.488,0,0,1,1.762.267,3.983,3.983,0,0,1,1.363.761,3.4,3.4,0,0,1,.877,1.2,3.823,3.823,0,0,1,.312,1.568,3.964,3.964,0,0,1-.245,1.462,4.114,4.114,0,0,1-.611,1.074,4.708,4.708,0,0,1-.807.789l-.828.628a6.378,6.378,0,0,0-.672.578,1.139,1.139,0,0,0-.34.639l-.255,1.624h-1.88l-.188-1.812a1.478,1.478,0,0,1,.149-.917,2.815,2.815,0,0,1,.579-.724,7.592,7.592,0,0,1,.806-.634,7.319,7.319,0,0,0,.834-.667,3.487,3.487,0,0,0,.65-.811,2.087,2.087,0,0,0,.261-1.074,1.643,1.643,0,0,0-.15-.717,1.617,1.617,0,0,0-.416-.544,1.853,1.853,0,0,0-.64-.35,2.559,2.559,0,0,0-.806-.122,3.565,3.565,0,0,0-1.072.139,3.867,3.867,0,0,0-.746.311c-.2.115-.376.219-.517.31a.707.707,0,0,1-.378.14.607.607,0,0,1-.578-.334Zm2.1,13.108a1.714,1.714,0,0,1,.127-.656,1.6,1.6,0,0,1,.351-.533,1.7,1.7,0,0,1,1.2-.49,1.636,1.636,0,0,1,.662.134,1.758,1.758,0,0,1,.534.356,1.628,1.628,0,0,1,.361.533,1.657,1.657,0,0,1-.895,2.2,1.708,1.708,0,0,1-.662.128,1.755,1.755,0,0,1-.667-.128,1.582,1.582,0,0,1-.884-.878A1.707,1.707,0,0,1,14.815,23.538Z"
        fill="#fff"
      />
    </svg>
  </div>
  
</div>
`;
